packages/babel-plugin-bugfix-safari-id-destructuring-collision-in-function-expression/src/util.ts

```diff
@@ -18,16 +18,16 @@ export function shouldTransform(
   // On collision, `getOwnBinding` returns the param binding
   // with the id binding be registered as constant violation
   const paramNameBinding = path.scope.getOwnBinding(name);
-  const constantViolations = paramNameBinding.constantViolations;
-  if (constantViolations.length === 0) {
-    // the function scope has no such collided bindings
+  if (paramNameBinding === undefined) {
+    // Case 1: the function id is injected by babel-helper-name-function, which
+    // assigns `NOT_LOCAL_BINDING` to the `functionId` and thus not registering id
+    // in scope tracking
+    // Case 2: the function id is injected by a third party plugin which does not update the
+    // scope info
```

Comment:
If we are in Case 2, it means that the function didn't originally have a paramter with the same name of the new injected id?

e.g. does this work?
```js
let x = function ([a]) {};
```
```js
// plugin
Function(path) {
  path.node.id = t.identifier("a");
}
```

(if it doesn't it's ok - the plugin needs to be fixed)

Reply:
Yeah it works if the `path` is then requeued (as is shown in the https://github.com/babel/babel/pull/13910/files#diff-9c3e0b47cab12356380982d8d3c8b5c3bbf3e160e18448619c2b54e9106bcb03) so that the bugfix can have a chance to be executed again.